include/swift/SIL/SILConstants.h

```diff
@@ -182,6 +191,9 @@ class SymbolicValue {
     /// This can be an array, struct, tuple, etc.
     Aggregate,
 
+    /// This is an enum.
+    Enum,
```

Comment:
I'd suggest splitting this representation into two (eventual) things:  Enum and EnumWithPayload.  

In the Enum representation, the "value" union can just store the 'EnumElementDecl*' directly avoiding a bump pointer allocation.  In the EnumWithPayload representation, you'd have EnumSymbolicValue that stores the EnumElementDecl* along with the SymbolicValue for the payload.

Reply:
This would prevent an extra allocation for trivial enums cases like Optional.None and other c-like enums.